pkg/controller/jenkins/notifications/mailgun/mailgun.go

```diff
@@ -0,0 +1,115 @@
+package mailgun
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/event"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications/provider"
+
+	"github.com/mailgun/mailgun-go/v3"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	infoColor    = "blue"
+	warningColor = "red"
+	defaultColor = "gray"
+
+	content = `
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
+        "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html>
+<head></head>
+<body>
+		<h1 style="background-color: %s; color: white; padding: 3px 10px;">%s</h1>
+		<h3>%s</h3>
+		<table>
+			<tr>
+				<td><b>CR name:</b></td>
+				<td>%s</td>
+			</tr>
+			<tr>
+				<td><b>Phase:</b></td>
+				<td>%s</td>
+			</tr>
+		</table>
+		<h6 style="font-size: 11px; color: grey; margin-top: 15px;">Powered by Jenkins Operator <3</h6>
+</body>
+</html>`
+)
+
+// MailGun is a sending emails notification service
+type MailGun struct {
+	k8sClient k8sclient.Client
+	config    v1alpha2.Notification
+}
+
+// New returns instance of MailGun
+func New(k8sClient k8sclient.Client, config v1alpha2.Notification) *MailGun {
+	return &MailGun{k8sClient: k8sClient, config: config}
+}
+
+func (m MailGun) getStatusColor(logLevel v1alpha2.NotificationLevel) event.StatusColor {
+	switch logLevel {
+	case v1alpha2.NotificationLevelInfo:
+		return infoColor
+	case v1alpha2.NotificationLevelWarning:
+		return warningColor
+	default:
+		return defaultColor
+	}
+}
+
+// Send is function for sending directly to API
+func (m MailGun) Send(event event.Event) error {
```

Comment:
I would separate the concerns of sending from message creation to simplify code for readability and testing